src/components/analytics/AnalyticsCharts.jsx

```diff
@@ -12,32 +11,53 @@ import AnalyticsApiService from './data/service';
 // eslint-disable-next-line no-unused-vars
 export default function AnalyticsCharts(enterpriseId) {
   const [token, setToken] = useState(null);
+  const [tokenUsedOnce, setTokenUsedOnce] = useState(false);
   const [isLoading, setIsLoading] = useState(false);
   const [error, setError] = useState(null);
-  // Fetch token
+  const ref = useRef(null);
+  let tableauUrl = null;
+  if (configuration.TABLEAU_URL) {
+    tableauUrl = `${configuration.TABLEAU_URL}/views/enterpriseadminanalytics/enroll_dash`;
+  }
+  const options = {
+    height: 900,
+    width: 1250,
+    hideToolbar: true,
+  };
+  const getUrl = () => {
+    const parsed = url.parse(tableauUrl, true);
+    const { protocol, host, pathname } = parsed;
+    const query = '?:embed=yes&:comments=no&:toolbar=no&:refresh=yes';
+    if (!tokenUsedOnce && token) {
+      setTokenUsedOnce(true);
+      return `${protocol}//${host}/trusted/${token}${pathname}${query}`;
+    }
+    return `${protocol}//${host}${pathname}${query}`;
```

Comment:
What will happen here as we are not passing the token in this URL? Is it to show some kind of error?

Reply:
Yes. This will trigger an un-authorized message on the iframe.